pkg/api/volumeGroup.go

```diff
@@ -77,34 +112,83 @@ func (v *VolumeGroupPortal) UpdateVolumeGroup() {
 	}
 
 	vg.Id = id
-
-	result, err := UpdateVolumeGroupDBEntry(c.GetContext(v.Ctx), vg)
+	result, err := UpdateVolumeGroupDBEntry(ctx, vg)
 	if err != nil {
 		v.ErrorHandle("Update volume group failed", model.ErrorInternalServer, err)
 		return
 	}
 	// Marshal the result.
 	body, err := json.Marshal(result)
 	if err != nil {
-		v.ErrorHandle("Marshal volume group updated result failed", model.ErrorInternalServer, err)
+		v.ErrorHandle("Marshal volume group updated result failed",
+			model.ErrorInternalServer, err)
 		return
 	}
-
 	v.SuccessHandle(StatusOK, body)
+
+	// NOTE:The real volume group creation process.
+	// Volume group creation request is sent to the Dock. Dock will set
+	// volume group status to 'available' after volume group creation operation
+	// is completed.
+	if err = v.CtrClient.Connect(CONF.OsdsLet.ApiEndpoint); err != nil {
+		log.Error("When connecting controller client:", err)
+		return
+	}
+	defer v.CtrClient.Close()
+
+	opt := &pb.CreateVolumeGroupOpts{
+		Message: string(body),
+		Context: ctx.ToJson(),
+	}
+	if _, err = v.CtrClient.CreateVolumeGroup(context.Background(), opt); err != nil {
+		log.Error("Create volume group failed in controller service:", err)
+		return
+	}
+
 	return
 }
 
 func (v *VolumeGroupPortal) DeleteVolumeGroup() {
 	if !policy.Authorize(v.Ctx, "volume_group:delete") {
 		return
 	}
+	ctx := c.GetContext(v.Ctx)
 
-	err := DeleteVolumeGroupDBEntry(c.GetContext(v.Ctx), v.Ctx.Input.Param(":groupId"))
+	id := v.Ctx.Input.Param(":groupId")
+	vg, err := db.C.GetVolumeGroup(ctx, id)
 	if err != nil {
-		v.ErrorHandle("Delete volume group failed", model.ErrorInternalServer, err)
+		v.ErrorHandle("Delete volume group failed",
+			model.ErrorBadRequest, err)
+		return
+	}
+
+	if err = DeleteVolumeGroupDBEntry(c.GetContext(v.Ctx), id); err != nil {
+		v.ErrorHandle("Delete volume group failed",
+			model.ErrorInternalServer, err)
 		return
 	}
+
 	v.SuccessHandle(StatusAccepted, nil)
+
+	// NOTE:The real volume group deletion process.
+	// Volume group deletion request is sent to the Dock. Dock will remove
+	// volume group record after volume group deletion operation is completed.
+	if err = v.CtrClient.Connect(CONF.OsdsLet.ApiEndpoint); err != nil {
```

Comment:
We should let controller update the database, not Dock.

Reply:
Ok, I will update the code in next PR.